### DISCUSSION TITLE
Most related entries

### DISCUSSION THREAD

**Discussion by HannahDeWachter** [craftcms/cms#13161] [Q&A] [answered]
On the page of a news article we would like to show other related news articles based on 3 different Category groups. We would like to sort these related articles on "most related" so entries with the most categories in common. My code however orders the entries based on postDate, most recent first.

My code:
````
{% set categories = (entry.catergoryA.ids()|merge(entry.catergoryB.ids()))|merge(entry.catergoryC.ids()) %}
{% set news = craft.entries().section('news').limit(4).relatedTo(categories).id('not ' ~ entry.id).all() %}
````

I have tried to add `.orderBy('score')` but this seems to only take the first category.

Can someone give me a hand?

**Comment by brandonkelly** [comment]
There’s no built-in way to accomplish this yet, but it’s been done before with some custom code / Preparse fields. See https://craftcms.stackexchange.com/q/1091/9 for a couple pointers

**Comment by mmikkel** [comment, marked as answer]
The [Similar](https://plugins.craftcms.com/similar?craft4) plugin could be helpful for your use case:

```twig
{% set categories = entry.catergoryA.ids()|merge(entry.catergoryB.ids())|merge(entry.catergoryC.ids()) %}
{% set news = craft.similar.find({ element: entry, context: categories, criteria: craft.entries.section('news').limit(4) }) %}
```

With the above, the `news` variable would be an array of (maximum 4) entries related to the category IDs from those three category fields; automatically sorted by the number of relations (the entries will also have an additional `count` attribute, referencing the total number of relations for each entry).

**Reply by HannahDeWachter** [reply]
Hi @mmikkel, I will try out this plugin and let you know if it works for my case! :)

**Reply by HannahDeWachter** [reply]
This is exactly what I was looking for. Thank you!
